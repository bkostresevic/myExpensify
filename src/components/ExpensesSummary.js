import React from 'react' 
import numeral from 'numeral'
import {connect} from 'react-redux' 
import selectExpensesTotal from '../selectors/expenses-total' 
import selectExpenses from '../selectors/expenses'

export const ExpensesSummary = ({expensesCount, expensesTotal}) => {
    const expenseWord = expensesCount === 1 ? 'expense' : 'expenses'
    const formatedExpensesTotal = numeral((expensesTotal / 100)).format('$0,0.00')
    return (
        <div>
           <h2>Viewing {expensesCount} {expenseWord} totaling {formatedExpensesTotal}</h2>
        </div>
    )
}

const mapStateToProps = (state) => {
    const visibleExpenses = selectExpenses(state.expenses, state.filters)
    return {
      expensesCount: state.expenses.length,
      expensesTotal: selectExpensesTotal(visibleExpenses)
    }
}

export default connect(mapStateToProps)(ExpensesSummary)

