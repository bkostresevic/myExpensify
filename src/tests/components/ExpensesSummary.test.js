import React from 'react' 
import {shallow} from 'enzyme' 
import {ExpensesSummary} from '../../components/ExpensesSummary' 

test('should render ExpensesSummary correctly with one expense', () => {
   const wrapper = shallow(<ExpensesSummary expensesCount={1} ExpensesSummary={235}/>)
   expect(wrapper).toMatchSnapshot()
})

test('should render ExpensesSummary correctly with multiple expenses', () => {
    const wrapper = shallow(<ExpensesSummary expensesCount={23} ExpensesSummary={587982544}/>)
    expect(wrapper).toMatchSnapshot()
})