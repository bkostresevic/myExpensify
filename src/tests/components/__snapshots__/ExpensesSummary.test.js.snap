// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpensesSummary correctly with multiple expenses 1`] = `
<div>
  <h2>
    Viewing 
    23
     
    expenses
     totaling 
    $0.00
  </h2>
</div>
`;

exports[`should render ExpensesSummary correctly with one expense 1`] = `
<div>
  <h2>
    Viewing 
    1
     
    expense
     totaling 
    $0.00
  </h2>
</div>
`;
